packages/earth-map/src/components/clip-layer/ClipLayer.tsx

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2018-2020 National Geographic Society
+ *
+ * Use of this software does not constitute endorsement by National Geographic
+ * Society (NGS). The NGS name and NGS logo may not be used for any purpose without
+ * written permission from NGS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+import FileSaver from 'file-saver';
+import { IPlace } from 'modules/places/model';
+import React, { useEffect, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import ExportService from 'services/ExportService';
+import LayersService from 'services/LayersService';
+
+import {
+  AsyncSelect,
+  Card,
+  DropdownItem,
+  ReactSelect,
+  serializeFilters,
+  Spinner,
+  TitleHero,
+} from '@marapp/earth-shared';
+
+interface IProps {
+  place: Partial<IPlace>;
+  onCancel?: () => void;
+  groups?: string[];
+}
+
+export function ClipLayer(props: IProps) {
+  const { onCancel, place, groups } = props;
+  const { id, name, organization } = place;
+  const [saveError, setSaveError] = useState('');
+  const [childLayers, setChildLayers] = useState([]);
+  const { t } = useTranslation();
+  const { register, handleSubmit, formState, control, watch, setValue } = useForm({
+    mode: 'onChange',
+  });
+  const { dirty, isValid, isSubmitting } = formState;
+  const selectedPrimaryLayer = watch('primaryLayer');
+  const selectedChildLayer = watch('childLayer');
+
+  // When primaryLayer changes, clear errors and set child layers
+  useEffect(() => {
+    setSaveError('');
+    setChildLayers(selectedPrimaryLayer?.references || []);
+  }, [selectedPrimaryLayer]);
+
+  // Pre-select the first child layer
+  useEffect(() => {
+    setValue('childLayer', childLayers[0]);
```

Comment:
Nice job! 

Just for  safety, wouldn't it be worthy double-checking if childLayers is an array, and we can access it's first element?